Build failure: rosegarden / dssi

## Steps To Reproduce

1. Install rosegarden via whichever mechanism.

## Build log

```
error: builder for '/nix/store/1pympx0isa69kf9bvvs18s1ydva9k01w-dssi-1.1.1.drv' failed with exit code 2;
       last 25 log lines:
       > Making all in jack-dssi-host
       > make[2]: Entering directory '/build/dssi-1.1.1/jack-dssi-host'
       > gcc -DHAVE_CONFIG_H -I. -I..    -I../dssi  -I/nix/store/rckfjiqi7i7mf7a6lxibgh6qnz3nc4ab-alsa-lib-1.2.13-dev/include -I/nix/store/zky910y4gki1bjjd3ma7n6cx6mc1v4hm-liblo-0.32/include -I/nix/store/bjs9vkm4yszlqxv31894rxhmw04dryqa-libjack2-1.9.22-dev/include -g -O2 -Wall -c -o jack_dssi_host-jack-dssi-host.o `test -f 'jack-dssi-host.c' || echo './'`jack-ds
si-host.c
       > In file included from /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/unistd.h:25,
       >                  from /nix/store/rckfjiqi7i7mf7a6lxibgh6qnz3nc4ab-alsa-lib-1.2.13-dev/include/alsa/asoundlib.h:31,
       >                  from jack-dssi-host.c:41:
       > /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/features.h:197:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wcpp-Wcpp8;;]
       >   197 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
       >       |   ^~~~~~~
       > jack-dssi-host.c: In function 'main':
       > jack-dssi-host.c:1236:59: error: passing argument 4 of 'lo_server_thread_add_method' from incompatible pointer type [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
       >  1236 |     lo_server_thread_add_method(serverThread, NULL, NULL, osc_message_handler,
       >       |                                                           ^~~~~~~~~~~~~~~~~~~
       >       |                                                           |
       >       |                                                           int (*)(const char *, const char *, lo_arg **, int,  void *, void *)
       > In file included from /nix/store/zky910y4gki1bjjd3ma7n6cx6mc1v4hm-liblo-0.32/include/lo/lo.h:33,
       >                  from jack-dssi-host.c:60:
       > /nix/store/zky910y4gki1bjjd3ma7n6cx6mc1v4hm-liblo-0.32/include/lo/lo_serverthread.h:151:72: note: expected 'lo_method_handler' {aka 'int (*)(const char *, const char *, lo_arg **, int,  struct lo_message_ *, void *)'} but argument is of type 'int (*)(const char *, const char *, lo_arg **, int,  void *, void *)'
       >   151 |                                const char *typespec, lo_method_handler h,
       >       |                                                      ~~~~~~~~~~~~~~~~~~^
       > make[2]: *** [Makefile:340: jack_dssi_host-jack-dssi-host.o] Error 1
       > make[2]: Leaving directory '/build/dssi-1.1.1/jack-dssi-host'
       > make[1]: *** [Makefile:445: all-recursive] Error 1
       > make[1]: Leaving directory '/build/dssi-1.1.1'
       > make: *** [Makefile:294: all] Error 2
       For full logs, run 'nix log /nix/store/1pympx0isa69kf9bvvs18s1ydva9k01w-dssi-1.1.1.drv'.
error: 1 dependencies of derivation '/nix/store/3wdxxl5815cmys656zqbzsm63nbhmn1r-rosegarden-24.12.drv' failed to build
```

Full build failure log at https://gist.github.com/nipsy/57bc72022b0032d2c0989ce7a56feb9a.

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

## Notify maintainers

@7c6f434c  (Apparently you maintain every package I use!) @lebastr 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ds/dssi/dssi-liblo.patch
+++ b/pkgs/by-name/ds/dssi/dssi-liblo.patch
@@ -1,0 +1,22 @@
+diff --git a/jack-dssi-host/jack-dssi-host.c b/jack-dssi-host/jack-dssi-host.c
+index f32ae69..119fe5a 100644
+--- a/jack-dssi-host/jack-dssi-host.c
++++ b/jack-dssi-host/jack-dssi-host.c
+@@ -119,7 +119,7 @@ LADSPA_Data get_port_default(const LADSPA_Descriptor *plugin, int port);
+ void osc_error(int num, const char *m, const char *path);
+ 
+ int osc_message_handler(const char *path, const char *types, lo_arg **argv, int
+-		      argc, void *data, void *user_data) ;
++		      argc, struct lo_message_ *data, void *user_data) ;
+ int osc_debug_handler(const char *path, const char *types, lo_arg **argv, int
+ 		      argc, void *data, void *user_data) ;
+ 
+@@ -1919,7 +1919,7 @@ int osc_debug_handler(const char *path, const char *types, lo_arg **argv,
+ }
+ 
+ int osc_message_handler(const char *path, const char *types, lo_arg **argv,
+-                        int argc, void *data, void *user_data)
++                        int argc, struct lo_message_ *data, void *user_data)
+ {
+     int i;
+     d3h_instance_t *instance = NULL;

--- a/pkgs/by-name/ds/dssi/package.nix
+++ b/pkgs/by-name/ds/dssi/package.nix
@@ -27,6 +27,10 @@ stdenv.mkDerivation rec {
     sha256 = "0kl1hzhb7cykzkrqcqgq1dk4xcgrcxv0jja251aq4z4l783jpj7j";
   };
 
+  patches = [
+    ./dssi-liblo.patch
+  ];
+
   nativeBuildInputs = [
     autoconf
     automake

